src/vuestic-theme/vuestic-components/va-dropdown/VaDropdown.vue

```diff
@@ -0,0 +1,298 @@
+      const dRect = this.$refs.dropdown.getBoundingClientRect()
+      const dWidth = dRect.width
+      const dHeight = dRect.height
+
+      const previousTop = this.top
+      const previousLeft = this.left
+
+      let left = 0
+      let top = 0
+      switch (this.position) {
+        case 'T':
+          top = tTop - dHeight - this.offset
+          left = tLeft - (dWidth - tWidth) / 2
+          break
+        case 'B':
+          top = tTop + tHeight + this.offset
+          left = tLeft - (dWidth - tWidth) / 2
+          break
+        case 'L':
+          top = tTop - (dHeight - tHeight) / 2
+          left = tLeft - dWidth - this.offset
+          break
+        case 'R':
+          top = tTop - (dHeight - tHeight) / 2
+          left = tLeft + tWidth + this.offset
+          break
+        case 'TL':
+          top = tTop - dHeight - this.offset
+          left = tLeft
+          break
+        case 'TR':
+          top = tTop - dHeight - this.offset
+          left = tLeft - (dWidth - tWidth)
+          break
+        case 'BR':
+          top = tTop + tHeight + this.offset
+          left = tLeft - (dWidth - tWidth)
+          break
+        case 'BL':
+          top = tTop + tHeight + this.offset
+          left = tLeft
+          break
+        default:
+          top = tTop + tHeight - this.offset
+          left = tLeft - (dWidth - tWidth)
+          break
+      }
+      this.left = `${Math.max(left, 0)}px`
+      this.top = `${Math.max(top, 0)}px`
+      if (forse && (this.top !== previousTop || this.left !== previousLeft)) {
+        this.$nextTick(this.calculatePosition)
+      }
+    },
+    toggle () {
+      if (this.isVisible) {
+        this.hide()
+      } else {
+        this.show()
+      }
+    },
+    show () {
+      this.visible = true
+      this.$nextTick(() => {
+        this.calculatePosition(true)
+        this.registerOutsideClick()
+      })
+    },
+    hide () {
+      this.visible = false
+      this.killOutSideClick()
+    },
+    close (event) {
+      if (!this.targetIsInsideElement(event)) {
+        this.hide()
+      }
+    },
+    targetIsInsideElement (e) {
+      let target = e.target
+      if (target) {
+        do {
+          if (target === this.$refs.dropdown || target === this.targetElement) {
+            return true
+          }
+        } while ((target = target.parentNode))
+      }
+      return false
+    },
+    countNesting () {
+      let target = this.targetElement
+      let count = 0
+      while (target) {
+        if (target.classList && target.classList.contains('va-dropdown')) {
+          count += 1
+        }
+        target = target.parentNode
+      }
+      return count
+    },
+    registerOutsideClick () {
+      this.outsideClickListener = event => {
+        this.close(event)
+      }
+      document.addEventListener('click', this.outsideClickListener)
+    },
+    killOutSideClick () {
+      document.removeEventListener('click', this.outsideClickListener)
```

Comment:
What if have dropdown opened and then navigate away from that page? Will it remove the listener?